numpy/core/_add_newdocs_scalars.py

```diff
@@ -176,6 +176,46 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     Any Python object.
     """)
 
+add_newdoc_for_scalar_type('str_', ['str0', 'unicode_'],
```

Comment:
It's tempting to just not mention these `str0` etc types. No one should be using them, and documenting them might just encourage people.